packages/api/amplify_api/test/graphql_test.dart

```diff
@@ -293,42 +301,63 @@ void main() {
   });
 
   group('Subscriptions', () {
+    var fakePlatform = MockConnectivityPlatform();
+    const graphQLDocument = '''subscription MySubscription {
+        onCreateBlog {
+          id
+          name
+          createdAt
+        }
+      }''';
+
+    final subscriptionRequest =
+        GraphQLRequest<String>(document: graphQLDocument);
+
+    final mockSubscriptionEvent = {
+      'id': subscriptionRequest.id,
+      'type': 'data',
+      'payload': {'data': mockSubscriptionData},
+    };
+
     setUp(() {
+      _failCount = 0;
       mockWebSocketService = MockWebSocketService();
+      const subscriptionOptions =
+          GraphQLSubscriptionOptions(pollInterval: Duration(seconds: 1));
 
       mockWebSocketBloc = MockWebSocketBloc(
         config: testApiKeyConfig,
         authProviderRepo: getTestAuthProviderRepo(),
         wsService: mockWebSocketService!,
+        subscriptionOptions: subscriptionOptions,
       );
-      // TODO(equartey): Add reconnection logic tests
-      // final fakePlatform = MockConnectivityPlatform();
-      // ConnectivityPlatform.instance = fakePlatform;
-      // final connectivity = Connectivity();
-
-      // WebSocketConnection.httpClientOverride = mockHttpClient;
-      // WebSocketConnection.connectivityOverride = connectivity;
-      // WebSocketConnection.webSocketFactoryOverride = getMockWebSocketChannel;
+      fakePlatform = MockConnectivityPlatform();
+      ConnectivityPlatform.instance = fakePlatform;
+      final connectivity = Connectivity();
+
+      WebSocketBloc.pollClientOverride = mockHttpClient;
+      WebSocketBloc.connectivityOverride = connectivity;
```

Comment:
Can you pass these in the constructor? May be important to have DI when we split out the Dart parts into their own package.